### ISSUE TITLE
Out Of Memory Error

### ISSUE THREAD

**Issue by siliconprime-tam** [apollographql/apollo-android#3111]
Our Android apps integrate the lastest sdk 
    implementation("com.apollographql.apollo:apollo-runtime:2.5.6")
    implementation("com.apollographql.apollo:apollo-android-support:2.5.6")
We got the out of memory error. The logs on crashlytic:

Fatal Exception: java.lang.OutOfMemoryError
Could not allocate JNI Env

com.apollographql.apollo.internal.interceptor.ApolloCacheInterceptor.interceptAsync (ApolloCacheInterceptor.java:56) <------ 
com.apollographql.apollo.internal.interceptor.RealApolloInterceptorChain.proceedAsync (RealApolloInterceptorChain.java:36)
com.apollographql.apollo.internal.fetcher.CacheFirstFetcher$CacheFirstInterceptor.interceptAsync (CacheFirstFetcher.java:31)
com.apollographql.apollo.internal.interceptor.RealApolloInterceptorChain.proceedAsync (RealApolloInterceptorChain.java:36)
com.apollographql.apollo.internal.RealApolloCall.enqueue (RealApolloCall.java:151)

This device got the error is Redmi 4X, run android 7.

**Comment by martinbonnin** [comment]
Thanks for sending this. Do you have more details about what's happening on the device at that point in time? Do you know what query it is and the size of the response?

**Comment by siliconprime-tam** [comment]
Hi Martinbonnin,

We have a list users and The apps make the query data to get information.

query QueryUser($uid: String!){
    user(uid: $uid){
        uid
        public{
            displayName
            username
            photoURL
            bio
        }
        isBot
        referralData{
            url
            content
        }
        kinAccount{
            stellarAddress
        }
        insideGatherings{
            id
        }
    }
}

**Comment by martinbonnin** [comment]
That doesn't sound too bad. Can you reproduce the issue? If yes, you can debug the memory usage with the profiler: https://developer.android.com/studio/profile/memory-profiler

**Comment by siliconprime-tam** [comment]
Hi Martinbonnin,

It's hard to reproduce this issues. We just got the logs on Firebase Crashlytics and feedback from users. If i have more details and the profiler, i'll send you. 

Thanks

**Comment by martinbonnin** [comment]
Yep, these are the worst to investigate... It seems unlikely that this specific query is responsible for the OOM, that's why I'm asking. I can't really imagine this taking megabytes of memory unless there's something terribly wrong. Things you can look for:

* uptime when this happens (I think crashlytics logs this). If this only happens after a long time, it's most likely a memory leak.
* running the memory profiler even if there's no crash. That might give you some insights about where the bigger allocations are.

**Comment by martinbonnin** [comment]
@siliconprime-tam anything else I can help with here?
